### PR TITLE
doc: Mention that writeln! and println! always use LF

### LINKED ISSUE #34697: writeln! documentation should tell what kind of newline is used
As I was using writeln! to fill an in-memory buffer for data that explicitly wants a specific type of newlines (LF, fortunately), and I was wondering whether writeln! would always use LF, or would use the type of newline relevant to the platform (essentially, LF on Unix, CR/LF on Windows), and couldn't find the answer in https://doc.rust-lang.org/std/macro.writeln!.html.

(I got my answer on IRC, it's always LF)

There may be other places where such information is relevant (println! comes to mind).

Cc: @steveklabnik

### PR DESCRIPTION
**Pull request by glandium**
Fixes #34697 

I'm not really satisfied with the wording, but I didn't have a better idea. Suggestions welcome.

### CODE REVIEW

File: src/libstd/macros.rs [hunk 1]
```diff
@@ -98,7 +98,9 @@ macro_rules! print {
     ($($arg:tt)*) => ($crate::io::_print(format_args!($($arg)*)));
 }
 
-/// Macro for printing to the standard output, with a newline.
+/// Macro for printing to the standard output, with a newline. On all
```
Comment (GuillaumeGomez) [inline src/libstd/macros.rs]: "On every platforms" maybe?

Comment (alexcrichton) [inline src/libstd/macros.rs]: Colloquially to me I think "On all platforms" makes a little more sense than "on every platforms"

Comment (GuillaumeGomez) [inline src/libstd/macros.rs]: Then it's all good for me. :wink:

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+ rollup

**Pull request merged**